### PR TITLE
Random bits of cleanup ¯\_(ツ)_/¯

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

**Comment by segiddins** [comment]
This should pass now

File: lib/bundler/dsl.rb [hunk 1]
```diff
@@ -332,7 +332,20 @@ def normalize_options(name, version, opts)
...
+                  val.call
+                when String
+                  begin
+                    eval(val, TOPLEVEL_BINDING) # rubocop:disable Lint/Eval
+                  rescue
+                    true
+                  end
+                else
+                  true
+                end
```
Comment (indirect) [inline lib/bundler/dsl.rb]: we didn't support evaling strings before, so I don't think this shouldn't be in a random cleanup—it should be in a PR, with a changelog entry and an addition to the docs for `install_if`. also I feel like an indented case statement like this is gross, can you extract the evaluation into a method?

**Comment by indirect** [comment]
👍  once comments are addressed and the specs pass

File: lib/bundler/dsl.rb (hunk 1 above)
Comment (segiddins) [inline lib/bundler/dsl.rb]: I can move this commit to a separate PR, sure. I think this was actually a remnant of me experimenting with something else

**Comment by segiddins** [comment]
@indirect done, and fixed up a RubyGems warning as well

File: lib/bundler/dsl.rb (hunk 1 above)
Comment (indirect) [inline lib/bundler/dsl.rb]: you should actually remove this now that it's in a separate PR 😁

**Comment by indirect** [comment]
@homu r+

File: lib/bundler/dsl.rb (hunk 1 above)
Comment (segiddins) [inline lib/bundler/dsl.rb]: I did?

**Comment by segiddins** [comment]
@homu r=indirect

**Pull request merged**